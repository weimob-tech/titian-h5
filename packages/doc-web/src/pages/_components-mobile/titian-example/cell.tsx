import { TiButton, TiCell, TiSwitch, TiImage, TiIcon } from '@titian-design/mobile-react';
import clsx from 'clsx';
import React from 'react';

import styles from './index.module.scss';

export default function CellExample(): JSX.Element {
  const ImageList = [
    {
      id: 1,
      path: 'https://image-c-dev.weimobwmc.com/qa-On6X/8b97cd488593474ba4a8ccaa3c1a493f.png',
      fileType: 'image',
    },
    { id: 2, path: 'https://res.wx.qq.com/wxdoc/dist/assets/img/0.4cb08bb4.jpg', fileType: 'image' },
    {
      id: 3,
      path: 'https://image-c-dev.weimobwmc.com/qa-On6X/8b97cd488593474ba4a8ccaa3c1a493f.png',
      fileType: 'image',
    },
  ];
  return (
    <div className={clsx(styles.section, styles.cell)}>
      <TiCell title="标题文字" desc="请选择内容" arrow required />
      <TiCell title="标题文字">
        <TiSwitch slot="desc" />
      </TiCell>
      <TiCell title="标题文字" divider={false}>
        <div className={clsx(styles['cell-images'])} slot="desc">
          {ImageList.map(image => (
            <div key={image.id}>
              <TiImage width={36} height={36} radius={3} src={image.path} lazyLoad>
                <TiIcon title="预览" extClass={clsx(styles['image-icon'])} size="10" name="search" color="#fff" />
              </TiImage>
            </div>
          ))}
        </div>
      </TiCell>
      <div className={clsx(styles['cell-button'])}>
        <TiButton extStyle={{ width: '117px' }} variant="filled" size="medium">
          取消
        </TiButton>
        <TiButton extStyle={{ width: '117px' }} size="medium">
          确定
        </TiButton>
      </div>
    </div>
  );
}
