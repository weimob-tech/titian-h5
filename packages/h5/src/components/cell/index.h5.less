@import '../common/static/var';
@import '../common/static/utils';
@import '../common/static/mixins/index.less';

@cell: ~'@{project}-cell';

:host {
  display: contents;
}

.@{cell} {
  @protected-cell-padding-v: var(--protected-cell-padding-v, @cell-padding-v);
  @protected-cell-padding-h: var(--protected-cell-padding-h, @cell-padding-h);
  @protected-cell-text-color: var(--protected-cell-text-color, @cell-text-color);
  @protected-cell-title-text-color: var(--protected-cell-title-text-color, @cell-title-text-color);
  @protected-cell-label-text-color: var(--protected-cell-label-text-color, @cell-label-text-color);
  @protected-cell-desc-text-color: var(--protected-cell-desc-text-color, @cell-desc-text-color);
  @protected-cell-bg-color: var(--protected-cell-bg-color, @cell-bg-color);
  @protected-cell-right-icon-color: var(--protected-cell-right-icon-color, @cell-right-icon-color);
  @protected-cell-text-disabled-color: var(--protected-cell-text-disabled-color, @cell-text-disabled-color);
  @protected-cell-hover-bg-color: var(--protected-cell-hover-bg-color, @cell-hover-bg-color);
  @protected-cell-required-color: var(--protected-cell-required-color, @cell-required-color);

  display: flex;
  position: relative;
  box-sizing: border-box;
  justify-content: space-between;
  width: 100%;
  padding: @protected-cell-padding-v @protected-cell-padding-h;

  transition-property: background-color;
  transition-duration: 0.1s;
  transition-timing-function: linear;

  background-color: @protected-cell-bg-color;

  color: @protected-cell-text-color;
  line-height: @cell-line-height;
  word-break: @cell-word-break;

  --icon-line-height: 1;

  &-divider::after {
    .hairline(bottom, @neutral-color-6);

    right: @gap-28;
    left: @gap-28;
  }

  &-disabled {
    pointer-events: none;

    .@{cell}-title,
    .@{cell}-label,
    .@{cell}-desc,
    .@{cell}-right-icon {
      color: @protected-cell-text-disabled-color;
    }
  }

  &-clickable:active {
    background-color: @protected-cell-hover-bg-color;
  }

  &-icon-box,
  &-required-box {
    display: flex;
    align-items: center;
    min-height: 42px;
  }

  &-icon {
    margin-right: @cell-icon-margin-right;
  }

  &-title {
    color: @protected-cell-title-text-color;
    font-size: @cell-title-font-size;
    font-weight: @cell-title-font-weight;
    line-height: @cell-title-line-height;
  }

  &-label {
    color: @protected-cell-label-text-color;
    font-size: @cell-label-font-size;
    font-weight: @cell-label-font-weight;
    line-height: @cell-label-line-height;
  }

  &-required {
    margin-left: @cell-required-margin-left;

    --protected-icon-line-height: @T42;
    --icon-color: @protected-cell-required-color;
  }

  &-title-wrap,
  &-desc-wrap {
    display: flex;
    justify-content: center;
  }

  &-title-wrap {
    align-items: center;
    justify-content: flex-start;
    width: @cell-title-wrap-width;
    min-width: @cell-title-wrap-min-width;
  }

  &-desc-wrap {
    flex-direction: column;
    align-items: flex-end;
  }

  &-desc-content {
    display: inline-flex;
    align-items: center;
    min-height: 42px;
  }

  &-desc {
    color: @protected-cell-desc-text-color;
    font-size: @cell-desc-font-size;
    font-weight: @cell-desc-font-weight;
    line-height: @cell-desc-line-height;
  }

  &-sub-desc {
    box-sizing: border-box;
    width: 100%;

    color: @protected-cell-label-text-color;
    font-size: @T24;
    font-weight: @font-weight-400;
    line-height: @T36;
    text-align: right;
  }

  &-right-icon {
    margin-left: @cell-right-icon-margin-left;

    --icon-color: @protected-cell-right-icon-color;
  }
}
