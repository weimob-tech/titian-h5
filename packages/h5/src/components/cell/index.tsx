import { Component, Element, h, Prop } from '@stencil/core';
import { JSXBase } from '@stencil/core/internal';
import { BasicComponentAbstract } from '../common/basic/BasicComponent';
import { isString, stringToAttrStyle, addShadowRootStyle } from '../common/utils';
import { join, handle } from '../common/utils/namespace';
import addUnit from '../common/utils/suffix';

@Component({
  tag: 'ti-cell',
  styleUrls: {
    pc: 'index.pc.less',
    h5: 'index.h5.less',
  },
  shadow: true,
})
export class TiCell implements BasicComponentAbstract {
  @Prop() extCss = '';

  @Prop() extClass?: string;

  @Prop() titleWidth?: string;

  @Prop() extStyle?: string | JSXBase.HTMLAttributes<Record<string, unknown>>['style'];

  /**
   * 标题
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" />
   * @since 0.1.0
   * */
  @Prop() title: string | any;

  /**
   * 副标题，右侧描述信息
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" desc="副标题" />
   * @since 0.1.0
   * */
  @Prop() desc?: string;

  /**
   * 标题下面的内容
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" label="标题下面的内容" />
   * @since 0.1.0
   * */
  @Prop() label?: string;

  /**
   * 副标题下面的内容
   *
   * @type string
   * @default ''
   * @example
   * <TiCell subDesc="副标题下面的内容" />
   * @since 0.1.0
   * */
  @Prop() subDesc?: string;

  /**
   * 图标
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" icon="home" />
   * @since 0.1.0
   * */
  @Prop() icon?: string;

  /**
   * 右侧图标的尺寸
   *
   * @type number
   * @default 36
   * @example
   * <TiCell iconSize={36} />
   * @since 0.1.0
   */
  @Prop() iconSize?: string | number = 36;

  /**
   * 右侧图标
   *
   * @type string
   * @default ''
   * @example
   * <TiCell title="标题" rightIcon="checked" />
   * @since 0.1.0
   * */
  @Prop() rightIcon?: string;

  /**
   * 右侧图标的尺寸
   *
   * @type number
   * @default 28
   * @example
   * <TiCell right-icon-size="{{ 28 }}" />
   * @since 0.1.0
   */
  @Prop() rightIconSize?: string | number = 28;

  /**
   * 图标颜色
   *
   * @type string
   * @default ''
   * @example
   * <TiCell color="red" />
   * @since 0.1.0
   *
   */
  @Prop() color?: string;

  /**
   * 是否显示右侧箭头
   *
   * @type boolean
   * @default true
   * @example
   * <TiCell arrow="{{false}}" />
   * @since 0.1.0
   * */
  @Prop() arrow?: boolean = false;

  /**
   * 是否禁用单元格
   *
   * @type boolean
   * @default false
   * @example
   * <TiCell title="标题" disabled />
   * @since 0.1.0
   * */
  @Prop() disabled?: boolean = false;

  /**
   * 是否必须选择
   *
   * @type boolean
   * @default false
   * @example
   * <TiCell required />
   * @since 0.1.0
   * */
  @Prop() required?: boolean = false;

  /**
   * 是否显示分割线
   *
   * @type boolean
   * @default true
   * @example
   * <TiCell divider={false} title="没有分割线" />
   * @since 0.1.0
   */
  @Prop() divider?: boolean = true;

  /**
   * 是否开启点击反馈
   *
   * @type boolean
   * @default true
   * @example
   * <TiCell clickable />
   * @since 0.1.0
   * */
  @Prop() clickable?: boolean = false;

  /**
   * 内容块的横向排列方式
   *
   * @type string
   * @default center
   * @example
   * <TiCell alignItems="center" />
   * @since 0.1.0
   * */
  @Prop() alignItems: 'center' | 'flex-start' | 'flex-end' | 'baseline' | 'stretch' | 'start' | 'end';

  @Prop() titleClass?: string;

  @Prop() titleWrapClass?: string;

  @Prop() labelClass?: string;

  @Prop() descClass?: string;

  @Prop({ reflect: true }) useSubArrow?: boolean = false;

  /**
   * 点击单元格回调的方法
   *
   * @param event TouchEvent
   * @return
   *
   * @example
   * <TiCell onClick="onClick" />
   * @since 0.1.0
   * */
  // @Event({ bubbles: true, composed: true }) click: EventEmitter<Record<string, any>>;

  @Element() protected host: HTMLDivElement;

  completedStyles(extStyle = {}) {
    return { ...stringToAttrStyle(extStyle), alignItems: this.alignItems };
  }

  private getClassName(): string {
    const { disabled, divider, extClass, clickable } = this;
    const classList = [join('cell', { disabled, divider, clickable }), extClass];

    return classList.filter(Boolean).join(' ');
  }

  private completedArrow(): string {
    if (isString(this.rightIcon) && this.rightIcon) {
      return this.rightIcon;
    }
    return 'arrow-right';
  }

  private completedRightIcon() {
    const { desc, rightIcon, rightIconSize, arrow } = this;
    if (!desc) {
      return { 'display': 'flex', 'justify-content': 'flex-end' };
    }
    if (rightIcon || arrow) {
      return { paddingRight: addUnit(+rightIconSize + 4) };
    }
    return { 'display': 'flex', 'justify-content': 'flex-end' };
  }

  renderLeftIcon() {
    return (
      <div class={handle('cell', 'icon-box')}>
        {this.icon ? (
          <ti-icon name={this.icon} size={this.iconSize || 36} color={this.color} class={handle('cell', 'icon')} />
        ) : (
          <slot name="icon" />
        )}
      </div>
    );
  }

  renderArrowIcon() {
    if (this.useSubArrow) {
      return null;
    }
    if (this.rightIcon || this.arrow) {
      return (
        <ti-icon
          color={this.color}
          size={this.rightIconSize}
          name={this.completedArrow()}
          ext-class={handle('cell', 'right-icon')}
        />
      );
    }
    return <slot name="right-icon" />;
  }

  componentWillLoad() {
    addShadowRootStyle.call(this);
  }

  render() {
    return (
      <div
        aria-hidden="true"
        part={this.extClass}
        class={this.getClassName()}
        style={this.completedStyles(this.extStyle)}
      >
        <div
          part={this.titleWrapClass}
          class={`${handle('cell', 'title-wrap')} ${this.titleWrapClass || ''}`}
          aria-hidden="true"
          style={this.completedStyles()}
        >
          {this.renderLeftIcon()}
          <div>
            <div style={{ ...this.completedStyles(), display: 'flex' }}>
              {this.title ? (
                <div
                  part={this.titleClass}
                  class={`${handle('cell', 'title')} ${this.titleClass}`}
                  style={this.titleWidth ? { width: this.titleWidth } : {}}
                >
                  {this.title ?? null}
                </div>
              ) : (
                <slot name="title" />
              )}
              {this.required ? (
                <div class={handle('cell', 'required-box')}>
                  <ti-icon name="required" size={24} ext-class={handle('cell', 'required')} />
                </div>
              ) : null}
            </div>

            {this.label ? (
              <div part={this.labelClass} class={`${handle('cell', 'label')} ${this.labelClass}`}>
                {this.label ?? null}
              </div>
            ) : (
              <slot name="label" />
            )}
          </div>
        </div>
        <div class={handle('cell', 'desc-wrap')} style={{ alignItems: this.alignItems }}>
          <div class={handle('cell', 'desc-content')} aria-hidden="true">
            {this.desc ? (
              <div part={this.descClass} class={`${handle('cell', 'desc')} ${this.descClass}`}>
                {this.desc ?? null}
              </div>
            ) : (
              <slot name="desc" />
            )}
            {this.renderArrowIcon()}
          </div>
          {this.subDesc ? (
            <div style={this.completedRightIcon()} class={handle('cell', 'sub-desc')}>
              {this.subDesc ?? null}
            </div>
          ) : (
            <slot name="sub-desc" />
          )}
        </div>
      </div>
    );
  }
}
