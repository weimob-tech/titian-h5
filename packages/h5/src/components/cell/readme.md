# ti-cell

<!-- Auto Generated Below -->


## Properties

| Property        | Attribute         | Description      | Type                                                                | Default     |
| --------------- | ----------------- | ---------------- | ------------------------------------------------------------------- | ----------- |
| `alignItems`    | `align-items`     | 内容块的横向排列方式       | `"baseline" \| "center" \| "flex-end" \| "flex-start" \| "stretch"` | `undefined` |
| `arrow`         | `arrow`           | 是否显示右侧箭头         | `boolean`                                                           | `false`     |
| `clickable`     | `clickable`       | 是否开启点击反馈         | `boolean`                                                           | `true`      |
| `color`         | `color`           | 图标颜色             | `string`                                                            | `undefined` |
| `desc`          | `desc`            | 副标题，右侧描述信息       | `string`                                                            | `undefined` |
| `descClass`     | `desc-class`      |                  | `string`                                                            | `undefined` |
| `disabled`      | `disabled`        | 是否禁用单元格          | `boolean`                                                           | `false`     |
| `divider`       | `divider`         | 是否显示分割线          | `boolean`                                                           | `true`      |
| `extClass`      | `ext-class`       | 额外的类名，添加到根节点的元素上 | `string`                                                            | `undefined` |
| `extStyle`      | --                | 额外的样式            | `{}`                                                                | `{}`        |
| `icon`          | `icon`            | 图标               | `string`                                                            | `undefined` |
| `iconSize`      | `icon-size`       | 右侧图标的尺寸          | `number \| string`                                                  | `36`        |
| `label`         | `label`           | 标题下面的内容          | `string`                                                            | `undefined` |
| `labelClass`    | `label-class`     |                  | `string`                                                            | `undefined` |
| `required`      | `required`        | 是否必须选择           | `boolean`                                                           | `false`     |
| `rightIcon`     | `right-icon`      | 右侧图标             | `string`                                                            | `undefined` |
| `rightIconSize` | `right-icon-size` | 右侧图标的尺寸          | `number \| string`                                                  | `28`        |
| `subDesc`       | `sub-desc`        | 副标题下面的内容         | `string`                                                            | `undefined` |
| `title`         | `title`           | 标题               | `any`                                                               | `undefined` |
| `titleClass`    | `title-class`     |                  | `string`                                                            | `undefined` |
| `useSubArrow`   | `use-sub-arrow`   |                  | `boolean`                                                           | `false`     |


## Events

| Event   | Description | Type                                 |
| ------- | ----------- | ------------------------------------ |
| `click` | 点击单元格回调的方法  | `CustomEvent<{ [x: string]: any; }>` |


## Dependencies

### Used by

 - [ti-collapse-item](../collapse-item)
 - [ti-dropdown-item](../dropdown-item)

### Depends on

- [ti-icon](../icon)

### Graph
```mermaid
graph TD;
  ti-cell --> ti-icon
  ti-collapse-item --> ti-cell
  ti-dropdown-item --> ti-cell
  style ti-cell fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
