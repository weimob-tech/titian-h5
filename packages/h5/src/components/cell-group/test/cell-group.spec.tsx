import { newSpecPage } from '@stencil/core/testing';
import { TiCell } from '../../cell';
import { TiCellGroup } from '..';
import { TestContainer } from '../../common/test/container';
import { h } from '@stencil/core';

describe('单元格渲染测试', () => {
  it('多个单元格组合在一起', async () => {
    const page = await newSpecPage({
      components: [TiCell, TiCellGroup, TestContainer],
      template: () => (
        <test-container>
          <ti-cell-group title="单元格组标题">
            <ti-cell title="标题" desc="详细内容文字" />
            <ti-cell title="标题" desc="详细内容文字" />
            <ti-cell title="标题" desc="详细内容文字" />
          </ti-cell-group>

          <ti-cell-group title="单元格组标题" sub-title="单元格组副标题">
            <ti-cell title="标题" desc="详细内容文字" />
            <ti-cell title="标题" desc="详细内容文字" />
            <ti-cell title="标题" desc="详细内容文字" />
          </ti-cell-group>

          <ti-cell-group title="单元格组模式为卡片" mode="card">
            <ti-cell title="标题" desc="详细内容文字" />
            <ti-cell title="标题" desc="详细内容文字" />
            <ti-cell title="标题" desc="详细内容文字" />
          </ti-cell-group>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('使用 slot 自定义 title', async () => {
    const page = await newSpecPage({
      components: [TiCell, TiCellGroup],
      template: () => (
        <ti-cell-group>
          <div slot="title">单元格组标题</div>
          <ti-cell title="标题" desc="详细内容文字" />
          <ti-cell title="标题" desc="详细内容文字" />
          <ti-cell title="标题" desc="详细内容文字" />
        </ti-cell-group>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
