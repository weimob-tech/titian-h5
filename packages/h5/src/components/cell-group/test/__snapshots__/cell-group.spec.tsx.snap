// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`单元格渲染测试 使用 slot 自定义 title 1`] = `
<ti-cell-group>
  <mock:shadow-root>
    <div class="titian-cell-group undefined">
      <slot name="title"></slot>
      <div class="titian-cell-group- titian-cell-group-body">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <div slot="title">
    单元格组标题
  </div>
  <ti-cell>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-cell titian-cell-divider">
        <div aria-hidden="true" class="titian-cell-title-wrap">
          <div class="titian-cell-icon-box">
            <slot name="icon"></slot>
          </div>
          <div>
            <div style="display: flex;">
              <div class="titian-cell-title undefined">
                标题
              </div>
            </div>
            <slot name="label"></slot>
          </div>
        </div>
        <div class="titian-cell-desc-wrap">
          <div aria-hidden="true" class="titian-cell-desc-content">
            <div class="titian-cell-desc undefined">
              详细内容文字
            </div>
            <slot name="right-icon"></slot>
          </div>
          <slot name="sub-desc"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-cell>
  <ti-cell>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-cell titian-cell-divider">
        <div aria-hidden="true" class="titian-cell-title-wrap">
          <div class="titian-cell-icon-box">
            <slot name="icon"></slot>
          </div>
          <div>
            <div style="display: flex;">
              <div class="titian-cell-title undefined">
                标题
              </div>
            </div>
            <slot name="label"></slot>
          </div>
        </div>
        <div class="titian-cell-desc-wrap">
          <div aria-hidden="true" class="titian-cell-desc-content">
            <div class="titian-cell-desc undefined">
              详细内容文字
            </div>
            <slot name="right-icon"></slot>
          </div>
          <slot name="sub-desc"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-cell>
  <ti-cell>
    <mock:shadow-root>
      <div aria-hidden="true" class="titian-cell titian-cell-divider">
        <div aria-hidden="true" class="titian-cell-title-wrap">
          <div class="titian-cell-icon-box">
            <slot name="icon"></slot>
          </div>
          <div>
            <div style="display: flex;">
              <div class="titian-cell-title undefined">
                标题
              </div>
            </div>
            <slot name="label"></slot>
          </div>
        </div>
        <div class="titian-cell-desc-wrap">
          <div aria-hidden="true" class="titian-cell-desc-content">
            <div class="titian-cell-desc undefined">
              详细内容文字
            </div>
            <slot name="right-icon"></slot>
          </div>
          <slot name="sub-desc"></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-cell>
</ti-cell-group>
`;

exports[`单元格渲染测试 多个单元格组合在一起 1`] = `
<test-container>
  <!---->
  <ti-cell-group>
    <mock:shadow-root>
      <div class="titian-cell-group undefined">
        <div class="titian-cell-group titian-cell-group-title-wrap undefined">
          <text class="titian-cell-group titian-cell-group-title undefined">
            单元格组标题
          </text>
        </div>
        <div class="titian-cell-group- titian-cell-group-body">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
  </ti-cell-group>
  <ti-cell-group sub-title="单元格组副标题">
    <mock:shadow-root>
      <div class="titian-cell-group undefined">
        <div class="titian-cell-group titian-cell-group-title-wrap undefined">
          <text class="titian-cell-group titian-cell-group-title undefined">
            单元格组标题
          </text>
          <text class="titian-cell-group titian-cell-group-sub-title undefined">
            单元格组副标题
          </text>
        </div>
        <div class="titian-cell-group- titian-cell-group-body">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
  </ti-cell-group>
  <ti-cell-group>
    <mock:shadow-root>
      <div class="titian-cell-group undefined">
        <div class="titian-cell-group titian-cell-group-title-wrap undefined">
          <text class="titian-cell-group titian-cell-group-title undefined">
            单元格组模式为卡片
          </text>
        </div>
        <div class="titian-cell-group-body titian-cell-group-card">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
    <ti-cell>
      <mock:shadow-root>
        <div aria-hidden="true" class="titian-cell titian-cell-divider">
          <div aria-hidden="true" class="titian-cell-title-wrap">
            <div class="titian-cell-icon-box">
              <slot name="icon"></slot>
            </div>
            <div>
              <div style="display: flex;">
                <div class="titian-cell-title undefined">
                  标题
                </div>
              </div>
              <slot name="label"></slot>
            </div>
          </div>
          <div class="titian-cell-desc-wrap">
            <div aria-hidden="true" class="titian-cell-desc-content">
              <div class="titian-cell-desc undefined">
                详细内容文字
              </div>
              <slot name="right-icon"></slot>
            </div>
            <slot name="sub-desc"></slot>
          </div>
        </div>
      </mock:shadow-root>
    </ti-cell>
  </ti-cell-group>
</test-container>
`;
